ftl/node/builder.py

```diff
@@ -73,29 +77,29 @@ def CreatePackageBase(self, base_image, cache, use_cache=True):
             else:
                 logging.info('No cached dependency layer for %s' % checksum)
         else:
-            logging.info('Skipping checking cache for dependency layer %s'
-                         % checksum)
-
-        layer, sha = self._gen_package_tar()
+            logging.info(
+                'Skipping checking cache for dependency layer %s' % checksum)
 
+        layer, sha = self._gen_package_tar(destination_path)
         with append.Layer(
-          base_image, layer, diff_id=sha, overrides=overrides) as dep_image:
+                base_image, layer, diff_id=sha,
+                overrides=overrides) as dep_image:
             if use_cache:
                 logging.info('Storing layer %s in cache.', sha)
                 cache.Store(base_image, _NODE_NAMESPACE, checksum, dep_image)
             else:
                 logging.info('Skipping storing layer %s in cache.', sha)
             return dep_image
 
-    def _gen_package_tar(self):
-        # We want the node_modules directory rooted at /app/node_modules in
-        # the final image.
+    def _gen_package_tar(self, destination_path):
+        # We want the node_modules directory rooted at:
+        # $destination_path/node_modules in the final image.
         # So we build a hierarchy like:
-        # /$tmp/app/node_modules
+        # /$tmp/$destination_path/node_modules
         # And use the -C flag to tar to root the tarball at /$tmp.
         tmp = tempfile.mkdtemp()
-        app_dir = os.path.join(tmp, 'app')
-        os.mkdir(app_dir)
+        app_dir = os.path.join(tmp, destination_path.strip("/"))
```

Comment:
Do you need strip with os.path.join?

Reply:
I believe so from:
'''
aprindle@aprindle-glaptop0:~$ python
Python 2.7.6 (default, Oct 26 2016, 20:30:19) 
[GCC 4.8.4] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.path.join("test", "test")
'test/test'
>>> os.path.join("test", "/test")
'/test'
'''